packages/lit-dev-content/src/components/local-time-converter.ts

```diff
@@ -0,0 +1,101 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {LitElement, html} from 'lit';
+import {customElement, property, state} from 'lit/decorators.js';
+
+/**
+ * Renders the localized date given an Intl.DateTimeFormatOptions format.
+ */
+@customElement('local-time-converter')
+export class LocalTimeConverter extends LitElement {
+  @property({
+    converter: {
+      fromAttribute: (value: string) => new Date(value),
+      toAttribute: (value: Date) => value.toString(),
+    },
+  })
+  date: Date = new Date();
+
+  @property({
+    attribute: 'format-weekday',
+  })
+  formatWeekday: 'narrow' | 'long' | 'short' | undefined = undefined;
+
+  @property({
+    attribute: 'format-year',
+  })
+  formatYear: 'numeric' | '2-digit' | undefined = undefined;
+
+  @property({
+    attribute: 'format-month',
+  })
+  formatMonth: 'narrow' | 'long' | 'short' | 'numeric' | '2-digit' | undefined =
+    undefined;
+
+  @property({
+    attribute: 'format-day',
+  })
+  formatDay: 'numeric' | '2-digit' | undefined = undefined;
+
+  @property({
+    attribute: 'format-time',
+  })
+  formatTime: 'full' | 'long' | 'medium' | 'short' | undefined = undefined;
+
+  @property({
+    attribute: 'format-time-zone',
+  })
+  formatTimeZone: 'long' | 'short' | undefined = undefined;
+
+  @state()
+  private targetLocale = 'en-US';
+
+  override render() {
+    console.log(this.targetLocale);
+    const dateFormat = new Intl.DateTimeFormat(this.targetLocale, {
+      weekday: this.formatWeekday,
+      year: this.formatYear,
+      month: this.formatMonth,
+      day: this.formatDay,
+      timeZoneName: this.formatTimeZone,
+    });
+
+    // timeFormat and timeZoneName cannot be used in the same format.
+    const timeFormat = new Intl.DateTimeFormat(this.targetLocale, {
+      timeStyle: this.formatTime,
+    });
+    const formattedDate = dateFormat.format(this.date);
+    // if no time format is specified, don't render the time.
+    const formattedTime = this.formatTime
+      ? ` ${timeFormat.format(this.date)}`
+      : '';
+    return html`<time part="root">${formattedDate}${formattedTime}</time>`;
+  }
+
+  firstUpdated() {
+    this.targetLocale = window?.navigator?.language || 'en-US';
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    window?.addEventListener('languagechange', this._languageChangeHandler);
```

Comment:
Why is window optional chained, and does a removeEventListener need to be implemented in the disconnectedCallback?

Reply:
was trying to be SSR safe. actually just gonna remove this component